### PR TITLE
Fix inaccurate documentation for TargetHttpsProxy resource

### PR DESCRIPTION
**Pull request by Hamzawy63**
This PR 
- Fixes inaccurate documentation for certificateManagerCertificates and certificateMap fields in TargetHttpsProxy 
- Fixes inaccurate documentation for certificateManagerCertificates field in RegionTargetHttpsProxy
- Addresses documentation issue referenced in https://github.com/hashicorp/terraform-provider-google/issues/17176

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: updated documentation of google_compute_target_https_proxy and google_compute_region_target_https_proxy
```

### CODE REVIEW

**Review by Stanwise — APPROVED** [review]

File: mmv1/products/compute/TargetHttpsProxy.yaml [hunk 1]
```diff
@@ -134,7 +134,8 @@ properties:
     name: 'certificateManagerCertificates'
     description: |
       URLs to certificate manager certificate resources that are used to authenticate connections between users and the load balancer.
-      Currently, you may specify up to 15 certificates. Certificate manager certificates do not apply when the load balancing scheme is set to INTERNAL_SELF_MANAGED.
+      Currently, you may specify up to 15 certificates. Certificate manager certificates only apply when the load balancing scheme is set to INTERNAL_MANAGED.
```
Comment (pawelJas) [inline mmv1/products/compute/TargetHttpsProxy.yaml:137]: "Currently, you may specify up to 15 certificates." this is not true for CertificateManagerCertificates.

```
URLs to Certificate manager certificate resources that are used to authenticate connections between users and the load balancer.
      Certificate manager certificates only apply when the load balancing scheme is set to INTERNAL_MANAGED.
      SslCertificates and CertificateManagerCertificates can not be defined together.
      ```

**Review by pawelJas — COMMENTED** [review]
Please also update RegionTargetHttpsProxy - The 15 limit is not true and also regional INTERNAL_SELF_MANAGED doesn't exists so it can be dropped.

Suggestion:
```
URLs to Certificate manager certificate resources that are used to authenticate connections between users and the load balancer.
SslCertificates and CertificateManagerCertificates can not be defined together.
Accepted format is `//certificatemanager.googleapis.com/projects/{project}/locations/{location}/certificates/{resourceName}` or just the self_link `projects/{project}/locations/{location}/certificates/{resourceName}`
```

**Review by melinath — CHANGES_REQUESTED** [review]
Removing from my queue pending the changes suggested above; I will rely on y'all to let me know when these docs are accurate & then merge.

File: mmv1/products/compute/TargetHttpsProxy.yaml (hunk 1 above)
Comment (Hamzawy63) [reply mmv1/products/compute/TargetHttpsProxy.yaml]: Done! Just to confirm, the 15 certificate limit is still true for sslCertificates and not for certificate manager certificate, right?

Comment (pawelJas) [reply mmv1/products/compute/TargetHttpsProxy.yaml]: Yes, exactly.

File: mmv1/products/compute/RegionTargetHttpsProxy.yaml [hunk 2]
```diff
@@ -140,8 +140,7 @@ properties:
     name: 'certificateManagerCertificates'
     description: |
       URLs to certificate manager certificate resources that are used to authenticate connections between users and the load balancer.
-      Currently, you may specify up to 15 certificates. Certificate manager certificates do not apply when the load balancing scheme is set to INTERNAL_SELF_MANAGED.
-      sslCertificates and certificateManagerCertificates fields can not be defined together.
+      sslCertificates and certificateManagerCertificates can not be defined together.
```
Comment (melinath) [inline mmv1/products/compute/RegionTargetHttpsProxy.yaml:143]: https://developers.google.com/style/contractions#negation-contractions

```suggestion
      sslCertificates and certificateManagerCertificates can't be defined together.
```

**Review by melinath — CHANGES_REQUESTED** [review]
a couple minor nits based on the dev docs style guide.

**Review by pawelJas — APPROVED** [review]

File: mmv1/products/compute/RegionTargetHttpsProxy.yaml (hunk 2 above)
Comment (Hamzawy63) [reply mmv1/products/compute/RegionTargetHttpsProxy.yaml]: Done!

File: mmv1/products/compute/TargetHttpsProxy.yaml [hunk 3]
```diff
@@ -164,7 +165,8 @@ properties:
     name: 'certificateMap'
     description: |
       A reference to the CertificateMap resource uri that identifies a certificate map
-      associated with the given target proxy. This field can only be set for global target proxies.
+      associated with the given target proxy. This field is only supported for EXTERNAL and EXTERNAL_MANAGED load balancing schemes.
+      For INTERNAL_MANAGED, please use certificate_manager_certificates instead.
```
[the remark this answers is not in the document]
Comment (Hamzawy63) [reply mmv1/products/compute/TargetHttpsProxy.yaml]: Done!

File: mmv1/products/compute/TargetHttpsProxy.yaml [hunk 4]
```diff
@@ -134,7 +134,8 @@ properties:
     name: 'certificateManagerCertificates'
     description: |
       URLs to certificate manager certificate resources that are used to authenticate connections between users and the load balancer.
-      Currently, you may specify up to 15 certificates. Certificate manager certificates do not apply when the load balancing scheme is set to INTERNAL_SELF_MANAGED.
+      Certificate manager certificates only apply when the load balancing scheme is set to INTERNAL_MANAGED.
+      For EXTERNAL and EXTERNAL_MANAGED please use certificate_map instead.
```
[the remark this answers is not in the document]
Comment (Hamzawy63) [reply mmv1/products/compute/TargetHttpsProxy.yaml]: Done

**Review by melinath — APPROVED** [review]
LGTM, going ahead with merge since @pawelJas has also approved

**Comment by melinath** [comment]
failing test is unrelated

**Pull request merged**